UI/_setup.test.ts

```diff
@@ -26,9 +27,9 @@ export async function test_db_view() {
     const relays = new Map<string, string[]>();
```

Comment:
change to `Map<string, Set<string>>` so that you don't need to de-dup in the implementation.

A better data strucutre makes your code simpler.